### ISSUE TITLE
Long combo boxes do not draw properly

### ISSUE THREAD

**Issue by spl237** [WayfireWM/wayfire#1620]
**Describe the bug**
When using Wayfire as the compositor, GTK combo box dropdowns do not display correctly when they have large numbers of items in them

**To Reproduce**
Build the following simple test case:

```
#include <stdio.h>
#include <stdlib.h>
#include <gtk/gtk.h>

int main (int argc, char *argv[])
{
    gtk_init (&argc, &argv);
    GtkWidget *win = gtk_window_new (GTK_WINDOW_TOPLEVEL);
    GtkListStore *ls = gtk_list_store_new (1, G_TYPE_STRING);
    GtkCellRenderer *rend = gtk_cell_renderer_text_new ();
    int n;
    for (n = 0; n < 100; n++)
    {
        gtk_list_store_insert_with_values (ls, NULL, n, 0, "String", -1);
    }
    GtkWidget *cb = gtk_combo_box_new_with_model (GTK_TREE_MODEL (ls));
    gtk_cell_layout_pack_start (GTK_CELL_LAYOUT (cb), rend, FALSE);
    gtk_cell_layout_add_attribute (GTK_CELL_LAYOUT (cb), rend, "text", 0);
    gtk_container_add (GTK_CONTAINER (win), cb);
    gtk_widget_show_all (win);
    gtk_main ();
    return 0;
}

```
Suggested command line - gcc gtktest.c `pkg-config --cflags --libs gtk+-3.0`

Run the application, and click the combo box.

The dropdown is shown as a black window. This does not happen if the number of items in the dropdown is small (< 75 for me, but it appears to be monitor resolution dependent - that's with a 1920 x 1080 monitor), but does if the number of items is large - change the max value of n in the for loop to change the number of items.

There are two GDK errors shown at the command line when the combo box is clicked:

(a.out:1355): Gdk-CRITICAL **: 18:11:05.159: gdk_wayland_display_get_monitor_at_window: assertion 'GDK_IS_WAYLAND_WINDOW (window)' failed

(a.out:1355): Gdk-CRITICAL **: 18:11:05.159: gdk_window_get_origin: assertion 'GDK_IS_WINDOW (window)' failed

The second of these errors also occurs when using mutter on Wayland as the compositor, but the first does not - that appears specific to wlroots / Wayfire.

**Expected behavior**
The dropdown should be shown as a list of items against a white background, as normal.

It appears correctly when the application is run under X, or under Wayland with mutter as the compositor.

**Screenshots or stacktrace**
https://www.dropbox.com/s/bf06qrmt56h2ywv/20221116_18h18m24s_grim.png

**Wayfire version**
0.7.4-2 - bookworm Debian package, backported to bullseye

**Comment by soreau** [comment]
I'm pretty sure this is a wlroots issue. To build wayfire against 0.16.0, you can use the `track-wlroots` branch of wayfire and pass `-Duse_system_wlroots=disabled` to meson.

**Comment by spl237** [comment]
Confirmed as fixed with wlroots 0.16.0.
